apps/opencs/view/render/scenewidget.cpp

```diff
@@ -35,14 +39,33 @@ namespace CSVRender
 {
 
 RenderWidget::RenderWidget(QWidget *parent, Qt::WindowFlags f)
-    : QWidget(parent, f)
+    : QOpenGLWidget(parent, f)
     , mRootNode(0)
 {
-
-    osgViewer::CompositeViewer& viewer = CompositeViewer::get();
-
     osg::DisplaySettings* ds = osg::DisplaySettings::instance().get();
-    //ds->setNumMultiSamples(8);
+    ds->setNvOptimusEnablement(1);
+    ds->setStereo(false);
+    ds->setNumMultiSamples(8);
+
+    QSurfaceFormat format = QSurfaceFormat::defaultFormat();
+
+#ifdef OSG_GL3_AVAILABLE
+    format.setVersion(3, 2);
```

Comment:
We use OpenGL 2.1, so 3.2 Core is *definitely* going to cause problems. Always set the version to 2.1. There's no such thing as a 2.1 Core or 2.1 Compatibility profile as nothing had been deprecated yet, so there's no point calling `setProfile`.

Also, it's probably best to not set up a debug context when the function in `gldebug.h` doesn't tell you to, and then when you do, hook up the debug callback that header exposes, too.

Reply:
Thank you! Fixed in commit b10a551 . This part was from here https://github.com/openscenegraph/osgQt/blob/2cb70673a4e83a618290e7ee66d52402a94ec3f6/examples/osgviewerQt/osgviewerQt.cpp#L48 . I honestly don't know anything about OpenGL versions, so I was just trying to convert whatever is in osgQt to use with OpenMW.